controller-server/src/main/java/com/yahoo/vespa/hosted/controller/role/RoleMembership.java

```diff
@@ -37,7 +49,7 @@ public String toString() {
      * membership to a {@link RoleMembership}.
      */
     public interface Resolver {
-        RoleMembership membership();
+        RoleMembership membership(Principal user);
```

Comment:
I liked it better before, with the resolving inside the resolver. How about making it `membership(Optional<Principal>)`? That way the context can be set for the current system, as before.

Reply:
After discussion with @tokle we agreed the `Principal` should not be optional — it is a setup error if this filter is run without a `Principal` being set in the request.